Property 'charset' of type 'undefined' is not assignable to 'string' index type 'string'.

Typescript 4.5.4 reports these these errors on hoofd 1.4.0

```
node_modules/hoofd/dist/dispatcher/index.d.ts:35:13 - error TS2411: Property 'charset' of type 'undefined' is not assignable to 'string' index type 'string'.

35             charset?: undefined;
               ~~~~~~~

node_modules/hoofd/dist/dispatcher/index.d.ts:56:13 - error TS2411: Property 'charset' of type 'undefined' is not assignable to 'string' index type 'string'.

56             charset?: undefined;
               ~~~~~~~

node_modules/hoofd/dist/index.d.ts:22:9 - error TS2411: Property 'charset' of type 'undefined' is not assignable to 'string' index type 'string'.

22         charset?: undefined;
           ~~~~~~~
```

caused by the various `toStatic` functions in that file https://unpkg.com/browse/hoofd@1.4.0/dist/dispatcher/index.d.ts:

```ts
    toStatic: () => {
        lang: string;
        title: string;
        links: any[];
        scripts: any[];
        metas: ({
            charset: "utf-8";
            content?: undefined;
        } | {
            [x: string]: string;
            content: string;
            charset?: undefined;
        })[];
    };
```

In the Typescript playground: https://www.typescriptlang.org/play?ts=4.5.4#code/KYDwDg9gTgLgBAE2AYwDYEMrDsiA7AZ3mS3RmABEBLAsM5AC2CgC44AKASjgF4A+OAG8AUHDgBbYDHQE2IsWIDaINkShU8AcwC6qmOq0BuUQtx5y5vQc3GFOBpgJSA-GwCueJADMNwBLbgAX2FAwyA

--- a/src/dispatcher/index.ts
+++ b/src/dispatcher/index.ts
@@ -1,4 +1,6 @@
 import { createContext } from 'react';
+import { LinkOptions } from '../hooks/useLink';
+import { ScriptOptions } from '../hooks/useScript';
 import { Name, CharSet, HttpEquiv, Property } from '../types';
 import { isServerSide } from '../utils';
 
@@ -196,7 +198,7 @@ export const createDispatcher = () => {
           }
         : // istanbul ignore next
           undefined,
-    toStatic: () => {
+    toStatic: (): StaticPayload => {
       //  Will process the two arrays, taking the first title in the array and returning <title>{string}</title>
       //  Then do a similar for the meta's. (will also need to add links, and add a linkQueue). Note that both queues
       //  will need a reset to prevent memory leaks.
@@ -242,6 +244,16 @@ export const createDispatcher = () => {
       };
     },
   };
+};
+
+export type StaticPayload = {
+  lang?: string;
+  title?: string;
+  links?: Array<LinkOptions>;
+  scripts?: Array<ScriptOptions>;
+  metas?: Array<
+    { charset: string } | { [key: string]: string; content: string }
+  >;
 };
 
 const defaultDispatcher = createDispatcher();

--- a/src/hooks/useHead.ts
+++ b/src/hooks/useHead.ts
@@ -3,7 +3,7 @@ import { DispatcherContext, META, MetaPayload, TITLE } from '../dispatcher';
 import { isServerSide } from '../utils';
 import { MetaOptions } from './useMeta';
 
-interface HeadObject {
+export interface HeadObject {
   title?: string;
   language?: string;
   metas?: MetaOptions[];

--- a/src/index.ts
+++ b/src/index.ts
@@ -7,7 +7,7 @@ export * from './hooks/useScript';
 export * from './hooks/useMeta';
 export * from './hooks/useTitle';
 export * from './hooks/useTitleTemplate';
-export { useHead } from './hooks/useHead';
+export { useHead, HeadObject } from './hooks/useHead';
 export * from './types';
 
 export const toStatic = dispatcher.toStatic;

